gufe/custom_codecs.py

```diff
@@ -52,18 +58,35 @@ def is_openff_quantity_dict(dct):
     from_dict=lambda dct: pathlib.PosixPath(dct["path"]),
 )
 
+
 BYTES_CODEC = JSONCodec(
     cls=bytes,
     to_dict=lambda obj: {'latin-1': obj.decode('latin-1')},
     from_dict=lambda dct: dct['latin-1'].encode('latin-1'),
 )
 
+
 DATETIME_CODEC = JSONCodec(
     cls=datetime.datetime,
     to_dict=lambda obj: {'isotime': obj.isoformat()},
     from_dict=lambda dct: datetime.datetime.fromisoformat(dct['isotime']),
 )
 
+
+NPY_DTYPE_CODEC = JSONCodec(
+    cls=np.generic,
```

Comment:
can use `cls=None` here: `cls` is only used in default functions; for this codec, you've overridden all4 funds.

Reply:
I noticed that the np.bool types are getting squashed to bool if this isn't set, not sure what's going on here.